ui/components/app/snaps/snap-link-warning/snap-link-warning.js

```diff
@@ -35,8 +35,10 @@ import { useI18nContext } from '../../../../hooks/useI18nContext';
 export default function SnapLinkWarning({ isOpen, onClose, url }) {
   const t = useI18nContext();
 
-  const parsedUrl = url && new URL(url);
-  const urlParts = parsedUrl && url.split(parsedUrl.host);
+  const parsedUrl = new URL(url);
+  const isHTTPS = parsedUrl.protocol === 'https:';
+  const urlParts = url.split(parsedUrl.host);
```

Comment:
What do you think about moving this to a util function, together with the conditional below?

```ts
function getUrl(...) {
  // Do the thing
  if (isHttps) {
    return <>...</>;
  }

  return url;
}

Reply:
Refactored to move to a small sub component